python_modules/dagster-test/dagster_test/toys/repo_resources/some_other_graph.py

```diff
@@ -0,0 +1,16 @@
+from dagster import op, graph, schedule
+
+
+@op(required_resource_keys={"abc"}, config_schema={"efg": str})
+def other_foo(context):
+    context.log(context.resources.abc)
+
+
+@graph
```

Comment:
Ideally when I want a fully configured graph, but not with resources (basically graph + run config), so it can be used in repose to create jobs (by adding resources), then I see two options:
* `configured` ops used in graphs (con:  it's not a single graph run_config  as it separates the `op` configurations) 
* I can create a list of `(graph, run_config)` pairs so in the repository 
```
[g.to_job(resource_defs=prod_resources, run_config=rc) for (g, rc) in graph_and_run_confs ]
```
This feels a bit hacky.

---
Ideally I would like something like: 
```
# in graphs.py
configured_graph = graph.configured(run_config={...})
# in repositories 
[configured_graph.to_job(resource_defs=prod_resources) for configured_graph in all_configured_graphs]
```

Reply:
@szalai1 - got it, that makes a lot of sense.  I don't believe that your latter suggestion currently works, but we should be able to make it work.  I filed a couple of issues:
* https://github.com/dagster-io/dagster/issues/4551
* https://github.com/dagster-io/dagster/issues/4556

Reply:
thanks, configured makes sense. I think the other option could be something like 

graph.set_run_config(...)
graph.set_resources(...)

but it would make really hard to track which state is the graph in. _is it configured or resources set or what?_

Reply:
> but it would make really hard to track which state is the graph in. is it configured or resources set or what?

Yeah, if we can avoid an API that mutates state, I think it will be much easier to reason about.

Would you want to be able to attach the config at graph construction time?  E.g.

```
@graph(config={...})
def my_graph():
    ...
```

Reply:
tbf most our pipelines have a single config, so I could use that, but not sure it makes sense to attach configs to the graph during declaration, it would prevent reusability, but it depends on what kind of behavior do you want to encourage too,